test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -708,6 +723,42 @@ void ProtoConverter::writeHelperFunctions()
 				return false;
 		return true;
 	}
+
+	/// Accepts function selector, correct argument encoding, and length of invalid encoding and returns
+	/// the correct and incorrect abi encoding for calling the function specified by the function selector.
+	function createEncoding(bytes4 funcSelector, bytes memory argumentEncoding, uint invalidLengthFuzz)
+		internal pure returns (bytes memory, bytes memory)
+	{
+		bytes memory validEncoding = new bytes(4 + argumentEncoding.length);
+		// Ensure that invalidEncoding crops at least one and at most all bytes from correct encoding.
```

Comment:
We can leave it like this for now and see what happens, but be aware:
It might be that ``abi.encode`` always adds padding to the argument encoding and that on the other hand decoding will not require that padding - as a consequence cropping one byte might not be enough to ensure failure (i.e. we might need to crop at least 31 bytes). (Consider a function taking a ``bytes`` argument of length 3 - I think ``abi.encode`` might pad the three bytes with zeros to 32 bytes, but even without those zeros the encoding is probably still considered valid and will decode just fine)

But yeah - I'm not entirely sure about this, so we'd need to verify this - and it might be just fine to keep it like this and revisit once we get a false positive due to this.

Reply:
That's interesting. I will make a note of this. Would you rather I copy-paste your warning as a comment in the code?

Reply:
As you like :-). We could also just check whether this is indeed a problem right away...

Reply:
Actually, the proto fuzzer always initializes a `bytes` argument with a value that is at least 32 bytes in size. Perhaps, this is a different issue that needs fixing? i.e., we should be intializing variables of type `bytes` with literals that are shorter than 32 bytes?

The routine that provides the literal used to initialize a `bytes` or a `string` type variable is this

https://github.com/ethereum/solidity/blob/cd563e526a92721a1e9e1633f5e2c9e53e77b242/test/tools/ossfuzz/protoToAbiV2.h#L242-L245

The routine hashes a monotonically increasing counter using  `keccak256` and assigns  the resulting 32 byte hash value to a variable of `bytes` or `string` type. The initialized value is a string literal (that holds the hash value).

Reply:
Moving this discussion to an independent issue.

Reply:
I think it definitely makes sense to test longer and shorter bytes and string arrays, yes.